packages/dd-trace/src/appsec/standalone.js

```diff
@@ -17,27 +65,72 @@ function onSpanStart ({ span, fields }) {
   }
 }
 
+function onSpanInject ({ spanContext, carrier }) {
+  // do not inject trace and sampling if there is no appsec event
+  if (!hasOwn(spanContext._trace.tags, APPSEC_PROPAGATION_KEY)) {
+    for (const key in carrier) {
+      const lKey = key.toLowerCase()
+      if (lKey.startsWith('x-datadog')) {
+        delete carrier[key]
+      } else if (lKey === 'tracestate') {
+        const tracestate = TraceState.fromString(carrier[key])
+        tracestate.forVendor('dd', state => state.clear())
+        carrier[key] = tracestate.toString()
+      }
+    }
+  }
+}
+
+function onSpanExtract ({ spanContext, carrier }) {
+  // reset upstream priority if _dd.p.appsec is not found
+  if (!hasOwn(spanContext._trace.tags, APPSEC_PROPAGATION_KEY)) {
+    resetSampling(spanContext)
+  } else if (spanContext._sampling.priority !== USER_KEEP) {
+    spanContext._sampling.priority = USER_KEEP
+  }
+}
+
+function sample (span) {
+  const spanContext = span.context?.()
+  if (enabled && spanContext._trace?.tags) {
+    spanContext._trace.tags[APPSEC_PROPAGATION_KEY] = '1'
+
+    // TODO: ask. can we reset here sampling like this?
```

Comment:
pending todo here, maybe to ask before opening the PR?

Reply:
So can we ?